src/ctia/stores/es/store.clj

```diff
@@ -15,6 +20,23 @@
   [{:keys [conn]}]
   (es-conn/close conn))
 
+(defn all-pages-iteration
+  "Returns lazy iteration of consecutive calls to `query-fn` with pagination params.
+
+  Resulted data is a sequence of responses of a shape `{:data [,,,] :paging {:next {,,,}}}`"
+  ([query-fn] (all-pages-iteration query-fn {}))
+  ([query-fn {:keys [limit]
+              :or {limit default-limit}
+              :as params}]
+   (let [limit (if (> limit max-result-window)
+                 max-result-window
+                 limit)]
```

Comment:
```suggestion
   (let [limit (min limit max-result-window)]
```

> only if missing — fallback to the "very default" limit provided by ductile library.
I'm not sure about this behaviour. If `limit` is `>` than the default-max-result-window but less than the `max-result-window`, `default-max-result-window` is used

Reply:
I identify those cases:
* limit wasn't provided — set it to `ductile/default-limit` which is 100
* limit is provided and its value less than `ductile/max-result-window` — keep the value
* limit is provided but more than `ductile/max-result-window` — use max-result-window

PS: I can see your point after I wrote this ^
Probably it would be better to handle only the case when the limit isn't set in params and keep it as is overwise. WDYT?

Reply:
yes sounds good